lib/ssh/ssh_wrapper.rb

```diff
@@ -33,22 +32,32 @@ def exec_with_logs!(session, command)
   # @param [Object] host_path
   # @param [Object] user
   # @param [Object] ip
-  # @return [Object] recursion
+  # @return [Integer]
   def sftp_get(remote_path, host_path, user, ip)
-    system("echo \"get #{remote_path} #{host_path}\" | sftp #{user}@#{ip}")
-    sleep 5 # Timeout between commands to not be banned by sftp
-    sftp_get(remote_path, host_path, user, ip) unless $CHILD_STATUS.success?
+    5.times do
+      result = system("echo \"get #{remote_path} #{host_path}\" | sftp #{user}@#{ip}")
+      sleep 5 # Timeout between commands to not be banned by sftp
+
+      raise StandardError, 'Failed to retrieve file' if result.nil?
+
+      break if result
+    end
   end
 
   # @param [Object] host_path
   # @param [Object] remote_path
   # @param [Object] user
   # @param [Object] ip
-  # @return [Object] recursion
+  # @return [Integer]
   def sftp_put(host_path, remote_path, user, ip)
-    system("echo \"put #{host_path} #{remote_path}\" | sftp #{user}@#{ip}")
-    sleep 5 # Timeout between commands to not be banned by sftp
-    sftp_put(host_path, remote_path, user, ip) unless $CHILD_STATUS.success?
+    5.times do
```

Comment:
Maybe extract similar code in `sftp_get` in `sftp_put` in new method and call it inside sftp_put something like
```
def sftp_put
  sftp_command("echo \"put #{host_path} #{remote_path}\" | sftp #{user}@#{ip}")
end
 ```